apps/hyperdrive-trading/src/ui/token/TokenPickerTwo.tsx

```diff
@@ -111,3 +116,114 @@ export function TokenPickerTwo({
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const fuse = useMemo(() => {
+    return new Fuse(tokens, {
+      keys: ["tokenConfig.symbol", "tokenConfig.name"],
+      threshold: 0.4,
+    });
+  }, [tokens]);
+
+  const filteredTokens = searchString
+    ? fuse.search(searchString).map(({ item }) => item)
+    : tokens;
+
+  const activeToken = tokens.find(
+    ({ tokenConfig }) => tokenConfig?.address === activeTokenAddress,
+  );
+
+  return (
+    <div>
+      {label ? (
+        <label className="daisy-label">
+          <span className="daisy-label-text">{label}</span>
+        </label>
+      ) : undefined}
+      <div className="daisy-dropdown daisy-dropdown-end daisy-dropdown-bottom">
+        <button
+          className={classNames(
+            "daisy-btn daisy-btn-md flex h-9 min-h-9 items-center rounded-box bg-neutral pl-3 pr-1",
+          )}
+          onClick={(e) => {
+            e.preventDefault();
+          }}
+        >
+          <img
+            src={activeToken?.tokenConfig?.iconUrl}
+            className="h-5 rounded-full"
+          />{" "}
+          {activeToken?.tokenConfig?.symbol}
+          <ChevronDownIcon className="h-6 text-neutral-content" />
+        </button>
+        <div
+          tabIndex={0}
+          className="daisy-menu daisy-dropdown-content z-[1] mt-1 w-64 gap-2 rounded-lg bg-neutral p-2 shadow"
+        >
+          <div className="mb-2 flex gap-2">
+            <div className="daisy-input daisy-input-sm flex w-full items-center gap-2 rounded-md bg-base-200">
+              <input
+                ref={inputRef}
+                className="w-full shrink bg-transparent outline-none"
+                placeholder="Search"
+                value={searchString}
+                onChange={({ target }) => setSearchString(target.value)}
+              />
+              {searchString.length ? (
+                <button
+                  type="button"
+                  title="Reset search"
+                  onClick={() => {
+                    inputRef.current?.focus();
+                    setSearchString("");
+                  }}
+                  className="text-neutral-content transition-all hover:text-current"
+                >
+                  <XMarkIcon className="size-5 fill-current" />
+                </button>
+              ) : (
+                <MagnifyingGlassIcon className="size-5 fill-neutral-content" />
+              )}
+            </div>
+          </div>
+          <ul className="max-h-[40vh] overflow-y-auto">
+            {searchString.length && !filteredTokens.length ? (
+              <li className="flex h-8 items-center px-4">No matches</li>
+            ) : (
+              filteredTokens.map(({ tokenConfig }) => (
+                <li key={tokenConfig?.address}>
+                  <button
+                    onClick={(e) => {
+                      onChange(tokenConfig?.address);
+                      e.preventDefault();
+                    }}
```

Comment:
The daisy dropdown stays open as long as it or a descendant has focus, but since this is something users are likely to be done with after choosing a token, we can call `blur()` on the active element to force the dropdown to close.

We can also remove the `preventDefault()` here too, and on all buttons as long as they have `type="button"`.

```suggestion
                    type="button"
                    onClick={() => {
                      (document.activeElement as HTMLElement)?.blur();
                      onChange(tokenConfig?.address);
                    }}
```

Reply:
Awesome, didn't know this. Thank you.